### DISCUSSION TITLE
[ZK Stack] - Using eth_call with a state override set - is this supported?

### DISCUSSION THREAD

**Discussion by miracle2k** [zkSync-Community-Hub/zksync-developers#429] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.76.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I can't get a state override to work.

The return value is always "no bytes", as if you are calling an address that was not deployed.

This is Python, but don't let that distract you. The call is essentially this:

```
result = await w3.eth.call(
        {
            'to': Web3.to_checksum_address('0x00000000678e52367adc69fd067e5edd1d33e3bf'),
            'data': data
        },
        state_override={
            Web3.to_checksum_address('0x00000000678e52367adc69fd067e5edd1d33e3bf'): {
                "code": '0x0000008003000039000000400030043f0000000102200190000000120000c13d000000000201001900000009022001980000001a0000613d000000000101043b0000000a011001970000000b0110009c0000001a0000c13d0000000001000416000000000101004b0000001a0000c13d0000000201000039000000800010043f0000000c010000410000001d0001042e0000000001000416000000000101004b0000001a0000c13d00000020010000390000010000100443000001200000044300000008010000410000001d0001042e00000000010000190000001e000104300000001c000004320000001d0001042e0000001e000104300000000000000000000000020000000000000000000000000000004000000100000000000000000000000000000000000000000000000000fffffffc000000000000000000000000ffffffff0000000000000000000000000000000000000000000000000000000026121ff000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000080000000000000000000000000000000000000000000000000000000000000000000000000000000009fb27058d76ff28162a1f578bb72427bf4d236d7e741ac5d06da187b062581e7'
            }
        }
```

Note that the code was generated using zksolc v1.4.0, compiling this contract:

```
contract Q {
    function f() public returns (uint256) {
        return 2;
    }
}
```

### Expected Behavior

I receive `2` from the `eth_call`

### Current Behavior

I am getting:

```Could not decode contract function call with return data: b'', output_types: ['uint256']```

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
@miracle2k thanks for reaching out. Will let the team know and get back to you soon

**Comment by githubdoramon** [comment, marked as answer]
@miracle2k unfortunately we don't yet support state overrides

**Comment by FatTailSec** [comment]
This is heart Breaking :(

**Comment by cpb8010** [comment]
This is supported now!
